cpp/src/sampling/random_walks.cuh

```diff
@@ -386,11 +354,15 @@ struct random_walker_t {
   random_walker_t(raft::handle_t const& handle,
                   graph_t const& graph,
                   index_t num_paths,
-                  index_t max_depth)
+                  index_t max_depth,
+                  vertex_t v_padding_val = 0,
+                  weight_t w_padding_val = 0)
```

Comment:
Should we better pad with invalid_vertex_id<vertex_t>::value() and std::numeric_limits<weight_t>::max()?

Reply:
and 0 here can be interpreted in two different ways: vertex ID 0 and an indicator saying v_padding_val is not provided.

I think it is better to just fix the padding value or use std::optional to allow users to optionally provide padding values.

Reply:
A few notes: 
1. there's no `std::optional` on device.
2. there's no `std::optional` in Cython;
3. by now there have been extended discussions about the "golden numbers" for padding.

I don't mind changing padding (again), but at some point we need to freeze the requirements and implement them. In general, design decisions like that should be made _before_ implementation, not after. 

For the record, `num_vertices` is currently used as padding value for vertices (not `0`; default value of `0` says: "if padding is used, then use `num_vertices` as padding for vertexes; if non-zero, and padding is used, then the supplied `v_padding_val` is used). 

For the weight padding, `weight_t{0}` was specifically requested.

Reply:
I see.

I am not sure ```there's no std::optional on device.``` matters but ```there's no std::optional in Cython;``` surely matters.

Just a suggestion, and if there has been an extended discussion and some contexts I am not aware of, no complaint.